src/Components/Projects/Projects.tsx

```diff
@@ -1,7 +1,331 @@
-import React from 'react'
+// nadia.mohamed.taha166@gmail.com
+// @Password123!
+import axios from "axios";
+import { useContext, useEffect, useState } from "react";
+import NoData from "./../../Shared/NoData/NoData";
+import noData from "./../../assets/images/no-data.png";
+import Modal from "react-bootstrap/Modal";
+import { useNavigate } from "react-router-dom";
+import { AuthContext } from "./../../Context/AuthContext";
+import { ToastContext } from "../../Context/ToastContext";
+import { useForm } from "react-hook-form";
 
-export default function Projects() {
+const Projects: React.FC = () => {
+  const { baseUrl, requestHeaders }: any = useContext(AuthContext);
+  const { getToastValue }: any = useContext(ToastContext);
+  const [project, setProject] = useState({});
+  const [projectDetails, setProjectDetails] = useState({});
+  const [projects, setProjects] = useState([]);
+  const navigate = useNavigate();
+  let [itemId, setItemId]: any = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+
+  interface FormValues {
+    title: any;
+    description: string;
+  }
+  const {
+    register,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>();
+  // **********to use more than one modal in same component**********
+  const [modalState, setModalState] = useState("close");
+  // ********to close modal*******************
+  const handleClose = () => setModalState("close");
+  // ********to show view modal*******************
+  const showViewModal = (id: any) => {
+    console.log("proj", project);
+    setItemId(id);
+    setModalState("view-modal");
+    getPtojectDetails(id);
+  };
+  // ***********update modal******************
+  const showUpdateModal = (project: any) => {
+    setItemId(project.id);
+    setValue("title", project.title);
+    setValue("description", project.description);
+    setModalState("update-modal");
+  };
+  // ********to show delete modal*******************
+
+  const showDeleteModal = (itemId: any) => {
+    setItemId(itemId);
+    setModalState("delete-modal");
+  };
+  // **********get all projects*****************
+  const getAllProjectsList = () => {
+    axios
+      .get(`${baseUrl}/Project/manager`, { headers: requestHeaders })
+      .then((response) => {
+        console.log(response?.data[0].title);
+        setProjects(response?.data);
+      })
+      .catch((error) => {
+        getToastValue(
+          "error",
+          error?.response?.data?.message ||
+            "An error occurred. Please try again."
+        );
+      });
+  };
+  // **********navigate to add proj******************
+  const navigateToNew = () => {
```

Comment:
While this function is presently employed within a button, employing the Link element and applying button-like styles would be a more suitable approach. The Link element is designed for user navigation upon clicking.

The useNavigate function, on the other hand, is intended for programmatic route navigation (e.g., within an onSubmit handler), typically executed after specific code. However, in our scenario, the sole purpose is to navigate upon clicking, without any additional actions.

Therefore, opting for the Link element in this case would be a more straightforward and fitting choice.